### PR TITLE
Multi architecture test-webserver image

### PR DESCRIPTION
**Pull request by mkumatag**
Signed-off-by: Manjunath A Kumatagi <mkumatag@in.ibm.com>

This PR will enable to build test-webserver image for different architecture like amd64, ppc64le, arm etc..

### CODE REVIEW

**Comment by pwittrock** [comment]
/unassign

I don't know what this is, and am already overwhelmed with PRs to review

**Comment by mkumatag** [comment]
@luxas @ixdy PTAL at this PR

File: for-demos/test-webserver/Makefile [hunk 1]
```diff
@@ -1,13 +1,77 @@
...
+TEST_REGISTRY ?= gcr.io/k8s-authenticated-test
+
+# Architectures supported: amd64, arm, arm64, ppc64le and s390x
+ARCH ?= amd64
+
+ALL_ARCH = amd64 arm arm64 ppc64le s390x
+
+GOARM=7
+TEMP_DIR := $(shell mktemp -d)
+GOLANG_VERSION=1.7.5
```
Comment (luxas) [inline for-demos/test-webserver/Makefile]: bump version to 1.8.3

File: for-demos/test-webserver/Makefile [hunk 2]
```diff
@@ -1,13 +1,77 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
 
-test-webserver: test-webserver.go
-	CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -ldflags '-w' ./test-webserver.go
+.PHONY: all all-container all-push all-push clean
 
-container: test-webserver
-	sudo docker build --pull -t gcr.io/google_containers/test-webserver .
+TAG ?= e2e
```
Comment (luxas) [inline for-demos/test-webserver/Makefile]: Please use a `1.x` tag

File: for-demos/test-webserver/Makefile [hunk 3]
```diff
@@ -1,13 +1,77 @@
...
 
-push: container
-	gcloud docker -- push gcr.io/google_containers/test-webserver
+REGISTRY ?= gcr.io/google-containers
+TEST_REGISTRY ?= gcr.io/k8s-authenticated-test
```
Comment (luxas) [inline for-demos/test-webserver/Makefile]: @ixdy Do we even need this code? I don't think so.

File: for-demos/test-webserver/Makefile [hunk 4]
```diff
@@ -1,13 +1,77 @@
...
-	CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -ldflags '-w' ./test-webserver.go
+.PHONY: all all-container all-push all-push clean
 
-container: test-webserver
-	sudo docker build --pull -t gcr.io/google_containers/test-webserver .
+TAG ?= e2e
 
-push: container
-	gcloud docker -- push gcr.io/google_containers/test-webserver
+REGISTRY ?= gcr.io/google-containers
```
Comment (luxas) [inline for-demos/test-webserver/Makefile]: Please move to `gcr.io/kubernetes-e2e-images`... or what do you think @ixdy?

**Review by luxas — COMMENTED** [review]
Kind of LGTM, but I think this code can be minimized/compressed/simplified a little.

Also I'm starting to think that is it better to move this to `test/images` in the core repo?
Where is this image used?

**Comment by mkumatag** [comment]
@luxas This code is used in many places under `test/e2e`, I'm happy to move this code to `test/images` if you are all okay with that.

**Comment by luxas** [comment]
@mkumatag Please do. Open a new PR here in contrib to remove this code and a new one in core to add the source for this image.

Some time in the very bright future we'll break out the full `test` to a separate repo, but that's not on the table right now.

**Pull request closed without merging**